app/controllers/units_controller.rb

```diff
@@ -0,0 +1,35 @@
+class UnitsController < ApplicationController
+  def create
+    @system = System.find(params[:unit][:system_id])
```

Comment:
why happens if there is nothing in `params[:unit]` (you'll get a NoMethodError for nil). You should probably have this resource nested or find another way to designate the system